packages/react-components/src/components/ParticipantListButton.tsx

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  concatStyleSets,
+  ContextualMenuItemType,
+  DefaultButton,
+  IButtonProps,
+  IContextualMenuItem,
+  IContextualMenuProps,
+  IStyle,
+  Label,
+  mergeStyles,
+  Stack
+} from '@fluentui/react';
+import { UserFriendsIcon } from '@fluentui/react-northstar';
+import copy from 'copy-to-clipboard';
+import React from 'react';
+import { ParticipantList, ParticipantListProps } from './ParticipantList';
+import {
+  controlButtonLabelStyles,
+  controlButtonStyles,
+  defaultParticipantListContainerStyle
+} from './styles/ControlBar.styles';
+
+/**
+ * Props for ParticipantListButton component
+ */
+export interface ParticipantListButtonProps extends IButtonProps {
+  /**
+   * Whether the label is displayed or not.
+   * @defaultValue `false`
+   */
+  showLabel?: boolean;
+  /**
+   * Props of ParticipantList component
+   */
+  participantListProps?: ParticipantListProps;
+  /**
+   * Styles of ParticipantList container
+   */
+  participantListContainerStyle?: IStyle;
+  /**
+   * Call Invite URL
+   */
+  callInviteURL?: string;
+  /**
+   * CallBack to mute all remote participants
+   *
+   * @param userId - ID of user requesting to mute all remote participants
+   */
+  onMuteAll?: (userId?: string) => void;
+}
+
+/**
+ * Generates default remote participants subMenuprops for a ParticipantListButton if the props contain participantListProps.
+ *
+ * @param props ParticipantListButtonProps
+ * @returns an array of IContextualMenuItem
+ */
+const generateDefaultRemoteParticipantsSubMenuProps = (props: ParticipantListButtonProps): IContextualMenuItem[] => {
+  const { participantListProps, onMuteAll } = props;
+  const items: IContextualMenuItem[] = [];
+
+  if (participantListProps && participantListProps.participants.length > 0) {
+    const remoteParticipantListProps = { ...participantListProps };
+
+    if (participantListProps.myUserId) {
```

Comment:
ParticipantList is displaying all participants (local and remote), in our case we don't want the local user to appear in the list as they appear on their own in the 'In this call' menuItem

Reply:
Yeah, this has been an open item for discussion.

I think pushing this decision in either direction is preferable to here:

  1. Contoso (or the Composite) filters down and only passes in remote participants here.
  2. We add a flag `ParticipantListProps`, named `excludeMe`, which defaults to false. When set, the local participant is not listed. It'd be OK to set `excludeMe` here when passing in `ParticipantListProps`.

I prefer 2 since it captures the two ways in which participant list is already used across the two composites we have, and takes boilerplate work away from Contoso. 

Essentially, I'm trying to avoid `ParticipantListButton` ever reading anything from `ParticipantListProps`. It creates tight coupling between the two. Blindly overwriting some props as they are passed along is OK.